### ISSUE TITLE
What is gtest_out.npy?

### ISSUE THREAD

**Issue by WurmD** [skelemoa/synse-zsl#7]
In synse_resources/ntu_results/shift_10_r/ we have 
g_label.npy - 50906 NTU120 action labels
gtest.npy - 50906 ShiftGCN 256D features
and gtest_out.npy - a 50906x120 array between [-19.3308 39.108074] 

What is gtest_out.npy? The output of the ShiftGCN classification layer (can't be, it's 120d, and you said you trained ShiftGCN only on the 110 seen classes)? 
Why is the size 50906? (NTU120 has in total 52 940 samples, and only 302 are samples with missing skeletons)

**Comment by pranaygupta36** [comment]
gtest_out.npy is the classification layer output from shift-gcn. Although the model is trained using samples from 110 classes only, the output size of the classification layer has to be kept 120 dimensional to accommodate all the classes(for the generalized ZSL setting). During training the original labels(range 0-119) of the samples are used, only the samples from the unseen classes are not used. 

Samples from the test set of NTU-120 are present in gtest_out. NTU-120 has a total of 114480 samples and for the experiments we have used the CS split (for simplicity).

**Comment by WurmD** [comment]
Thank you for the clarification.

What is the CS split? (can't find a mention to it neither in https://github.com/shahroudy/NTURGB-D nor https://github.com/skelemoa/synse-zsl nor in your paper)

**Comment by pranaygupta36** [comment]
Basically, there are two types of train-test split suggested by the authors of the NTU-120 dataset [paper](https://arxiv.org/pdf/1905.04757.pdf). One is Cross-Subject(CS), the other is Cross Setup(CSet). We have created our zsl splits using Cross Subject(CS). I apologize CS, and CSet is actually the notation we use within our lab.

**Comment by WurmD** [comment]
Are you able to share the algorithm or the code used to create such files?

I'd like to know to which original skeleton files each line in the g_label gtest and gtest_out belong to.

**Comment by pranaygupta36** [comment]
Sure, I can share the data preparation code we used. Although it will take me sometime to upload it, as it is scattered around in a few jupyter notebooks in my local machine.

**Comment by WurmD** [comment]
*nudge* @pranaygupta36 :) 
This would be useful to make this repo complete (namely it would allow users to retrain SynSE; I was trying to retrain SynSE and hit another road block, as for the retraining of the gating model it expects `synse_10_r_unseen_zs.npy`, `synse_10_r_seen_zs.npy`, `ztest_out.npy`, and `val_out.npy`, all of each are not automatically generated by `synse_training.py`)

**Comment by pranaygupta36** [comment]
hey @WurmD sorry for the long delay, you are correct all the files are not automatically generated, the ztest_out.npy and the val_out.npy files are within the ' synse_resources/ntu_results/<experiment_name>' folders. They can be found in the 'val' experiment folder. For ex. synse_10_r_val

The val_out.npy contains the final layer softmax scores of the samples selected for validation and the ztest_out.npy are the final layer softmax scores of the zsl test samples. The scores are obtained from the skeleton feature encoder. If you can provide a bit more information about the dataset using which you are trying to retrain the network, I might be able to help you a little more.
